public/expressions/overlay_anomalies.ts

```diff
@@ -0,0 +1,193 @@
+//  * this file be licensed under the Apache-2.0 license or a
+//  * compatible open source license.
+//  *
+//  * Any modifications Copyright OpenSearch Contributors. See
+//  * GitHub history for details.
+//  */
+
+import { get } from 'lodash';
+import { i18n } from '@osd/i18n';
+import { ExpressionFunctionDefinition } from '../../../../src/plugins/expressions/public';
+import {
+  VisLayerTypes,
+  VisLayers,
+  ExprVisLayers,
+  PluginResource,
+} from '../../../../src/plugins/vis_augmenter/public';
+import {
+  TimeRange,
+  calculateBounds,
+} from '../../../../src/plugins/data/common';
+import {
+  getAnomalySummaryQuery,
+  parsePureAnomalies,
+} from '../pages/utils/anomalyResultUtils';
+import { AD_NODE_API } from '../../utils/constants';
+import { AnomalyData } from '../models/interfaces';
+import { getClient } from '../services';
+import {
+  PointInTimeEventsVisLayer,
+  VisLayerError,
+  VisLayerErrorTypes,
+} from '../../../../src/plugins/vis_augmenter/public';
+import { PLUGIN_NAME } from '../../public/utils/constants';
+import { NO_PERMISSIONS_KEY_WORD } from '../../server/utils/helpers';
+
+type Input = ExprVisLayers;
+type Output = Promise<ExprVisLayers>;
+
+interface Arguments {
+  detectorId: string;
+}
+
+const name = 'overlay_anomalies';
+
+export type OverlayAnomaliesExpressionFunctionDefinition =
+  ExpressionFunctionDefinition<'overlay_anomalies', Input, Arguments, Output>;
+
+// This gets all the needed anomalies for the given detector ID and time range
+const getAnomalies = async (
+  detectorId: string,
+  startTime: number,
+  endTime: number
+): Promise<AnomalyData[]> => {
+  const anomalySummaryQuery = getAnomalySummaryQuery(
+    startTime,
+    endTime,
+    detectorId,
+    undefined,
+    false
+  );
+
+  // We set the http client in the plugin.ts setup() fn. We pull it in here to make a
+  // server-side call directly.
+  // Note we can't use the redux fns here (e.g., searchResults()) since it requires
+  // hooks (e.g., useDispatch()) which doesn't make sense in this context, plus is not allowed by React.
+  const anomalySummaryResponse = await getClient().post(
+    `..${AD_NODE_API.DETECTOR}/results/_search`,
+    {
+      body: JSON.stringify(anomalySummaryQuery),
+    }
+  );
+
+  return parsePureAnomalies(anomalySummaryResponse);
+};
+
+// This takes anomalies and returns them as vis layer of type PointInTimeEvents
+const convertAnomaliesToLayer = (
+  anomalies: AnomalyData[],
+  ADPluginResource: PluginResource
+): PointInTimeEventsVisLayer => {
+  const events = anomalies.map((anomaly: AnomalyData) => {
+    return {
+      timestamp: anomaly.startTime + (anomaly.endTime - anomaly.startTime) / 2,
+      metadata: {},
+    };
+  });
+  return {
+    originPlugin: 'anomaly-detection',
+    type: VisLayerTypes.PointInTimeEvents,
+    pluginResource: ADPluginResource,
+    events: events,
+  } as PointInTimeEventsVisLayer;
+};
+
+export const overlayAnomaliesFunction =
```

Comment:
Just for further documentation, could you add a comment for this function describing more about its purpose / how it's adding a vislayer onto an existing array of vislayers? Making that more clear can help us reference it when adding more documentation / tutorials on how other plugins may integrate.

Reply:
Added